packages/components/resource-pages/list-page.tsx

```diff
@@ -1,31 +1,206 @@
 import * as React from 'react';
+import {
+  K8sResourceCommon,
+  ListPageBody,
+  ListPageCreateLink,
+  ListPageFilter,
+  TableColumn,
+  useActiveColumns,
+  useK8sWatchResource,
+  useListPageFilter,
+  VirtualizedTable,
+} from '@openshift-console/dynamic-plugin-sdk';
 import { K8sModel } from '@openshift-console/dynamic-plugin-sdk/lib/api/common-types';
 import * as _ from 'lodash';
-import { NooBaaBackingStoreModel, NooBaaBucketClassModel } from '../../models';
+import { useTranslation } from 'react-i18next';
+import { DATA_FEDERATION_NAMESPACE } from '../../constants';
+import {
+  NooBaaBackingStoreModel,
+  NooBaaBucketClassModel,
+  NooBaaNamespaceStoreModel,
+} from '../../models';
+import { referenceForModel } from '../../utils';
 import { CustomKebabItemsType } from '../../utils/kebab/kebab';
-import { ModalMap } from '../../utils/modals/modalLauncher';
+import {
+  LaunchModal,
+  ModalMap,
+  useModalLauncher,
+} from '../../utils/modals/modalLauncher';
+import {
+  RowRenderer as BPRowRenderer,
+  useBucketPolicyList,
+} from './bucket-policy-list-view';
+import {
+  RowRenderer as DRRowRenderer,
+  useDataResourceList,
+} from './data-resource-list-view';
+import './list-view.scss';
+
+const ModelRowRendererMap = {
+  [referenceForModel(NooBaaNamespaceStoreModel)]: DRRowRenderer,
+  [referenceForModel(NooBaaBucketClassModel)]: BPRowRenderer,
+};
+
+const ResourceTable: React.FC<ResourceTableProps> = ({
+  tableColumns,
+  resourceModel,
+  ...props
+}) => {
+  const { t } = useTranslation();
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null,
+  });
+
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t('Resource Page')}
+      columns={columns}
+      Row={ModelRowRendererMap[referenceForModel(resourceModel)]}
+    />
+  );
+};
+
+const GenericListPage: React.FC<GenericListPageProps> = ({
+  actions,
+  resourceModel,
+  kebabActions,
+  resourceMap,
+  tableColumns,
+  createButtonTitle,
+}) => {
+  const { t } = useTranslation();
+  const [Modal, modalProps, launchModal] = useModalLauncher(actions);
+
+  const resource = React.useMemo(
+    () => ({
+      kind: referenceForModel(resourceModel),
+      namespace: DATA_FEDERATION_NAMESPACE,
+      isList: true,
+    }),
+    [resourceModel]
+  );
+
+  const [resources, loaded, error] =
+    useK8sWatchResource<K8sResourceCommon[]>(resource);
```

Comment:
Please make sure,  when calling GenericListPage from multiple places with diff resources not to affect the useK8sWatchResource. I worried it may override the 
"resources, loaded, error".

Just a thought GenericListPage should be a async to avoid this problem. I dont know i am correct or wrong here. Please check once.

Reply:
Ah! ok, you won't call all pages same time so, yes it is safe. Please ignore this comment :)